### PR TITLE
Set paths correctly when loading newer versions

### PR DESCRIPTION
**Pull request by dodecaphonic**
When loading newer versions, installed under `NVM_HOME/versions`, the paths weren't being set correctly, as the code still built them assuming everything fell under `NVM_HOME`. This pull request fixes that, and also enables loading io.js versions.

### CODE REVIEW

**Comment by rejeep** [comment]
Can you please rebase to master so that Travis runs properly, thanks!

File: nvm.el [hunk 1]
```diff
@@ -4,10 +4,11 @@
 
 ;; Author: Johan Andersson <johan.rejeep@gmail.com>
 ;; Maintainer: Johan Andersson <johan.rejeep@gmail.com>
-;; Version: 0.1.0
+;; Version: 0.1.1
+;; Package-Version: 20150906.311
```
Comment (rejeep) [inline nvm.el]: This should not be here.

**Comment by rejeep** [comment]
Io.js is being merged with Node, so is it really a good idea to add support for it?

**Comment by dodecaphonic** [comment]
My rationale for supporting it was the possibility of someone having a project they haven't upgraded to node 4 yet. I personally don't have a need for it. If you wish,  I'll remove it.

File: nvm.el (hunk 1 above)
Comment (dodecaphonic) [inline nvm.el]: The version bump?

**Comment by dodecaphonic** [comment]
I've rebased it.

File: nvm.el [hunk 2]
```diff
@@ -54,35 +58,63 @@
...
+
+(defun nvm--version-name (runtime path)
+  "Makes runtime names match those in nvm ls"
+  (if (string= "node" runtime)
+      (f-filename path)
+    (concat (nvm--clean-runtime-name runtime) "-" (f-filename it))))
+
+(defun nvm--version-directories-new (match-fn)
+  (if (not (nvm--using-new-path-schema?))
+      nil
```
Comment (rejeep) [inline nvm.el]: Isn't this the same?

```lisp
(when (nvm--using-new-path-schema?)
   ;; ...
  )
```

File: nvm.el [hunk 3]
```diff
@@ -4,10 +4,10 @@
 
 ;; Author: Johan Andersson <johan.rejeep@gmail.com>
 ;; Maintainer: Johan Andersson <johan.rejeep@gmail.com>
-;; Version: 0.1.0
+;; Version: 0.1.1
```
Comment (rejeep) [inline nvm.el]: Can you please remove this. I usually do a separate commit with this and also release a tag for it.

**Comment by dodecaphonic** [comment]
Updated to remove the version bump and use your `when` suggestion (my elisp-fu is not so strong, thank you for that).

**Comment by dodecaphonic** [comment]
Do you want me to take out the io.js support?

**Comment by rejeep** [comment]
> Do you want me to take out the io.js support?

No, let's leave it as you already have it implemented. As you say, not everyone can update their Node version.

**Comment by rejeep** [comment]
Thanks for this pull request, merged! :+1: 

Btw, I did a minor release as you added support for IO.js.

**Comment by dodecaphonic** [comment]
Thank you for your guidance and patience. :smile:
